### PR TITLE
Add analyticsString to angular model close button

### LINKED ISSUE #3452: [bug] angular spark modal needs data analytics on the close button
Put the HTML attribute on the close button in the component and add an Input for it of `analyticsString`

### PR DESCRIPTION
**Pull request by ka6thick54**
## What does this PR do?
Adds analyticsString to Angular Modal Close button

### Associated Issue
Fixes #3452 

## Please check off completed items as you work.
If a checklist item or section does not apply to your PR then please remove it.

### Documentation
 - [ ] Update Spark Docs

### Code
 - [ ] Build Component in HTML
 - [ ] Build Component in Angular
 - [ ] Build Component in React
 - [ ] Unit Testing in HTML with `npm run test` in `html/` (100% coverage, 100% passing)
 - [ ] Unit Testing in Angular with `npm run test` in `angular/` (100% coverage, 100% passing)
 - [ ] Unit Testing in React with `npm run test` in `react/` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [ ] Google Chrome
  - [ ] Google Chrome (Mobile)
  - [ ] Mozilla Firefox
  - [ ] Mozilla Firefox (Mobile)
  - [ ] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [ ] Microsoft Edge
  - [ ] Apple Safari
  - [ ] Apple Safari (Mobile)
  
### Accessibility Testing
  - [ ] Axe browser extension
  - [ ] Jaws Inspect
  - [ ] VoiceOver (iOS) or Talkback (Android)

### Deploy Preview Reviewed and Approved
 - [ ] Design Team
 - [ ] Accessibility Team

### Screenshots
Add screenshots to help explain your PR if you'd like. However, this is not
expected.

### CODE REVIEW

**Comment by afebbraro** [comment]
Thank you so much! We will get this reviewed on Monday!

File: angular/projects/spark-angular/src/lib/components/sprk-modal/sprk-modal.component.ts [hunk 1]
```diff
@@ -132,6 +133,16 @@ export class SprkModalComponent {
    * per page.
    */
   @Input()
+  closeAnalyticsString: string;
```
Comment (guern1kn) [inline angular/projects/spark-angular/src/lib/components/sprk-modal/sprk-modal.component.ts:136]: @ka6thick54 Thanks for the PR! Let's update the description of this new Input to match the other `analyticsString` Inputs. 
/**
   * The value supplied will be assigned to the
   * `data-analytics` attribute
   * on the close button. Intended
   * for an outside library to capture data.
   */

**Review by guern1kn — CHANGES_REQUESTED** [review]
Thanks for the PR! I left a comment. Could you also add a unit test for the new Input? If you have any questions, please let us know!

**Comment by guern1kn** [comment]
@ka6thick54 If you don't have time to do the test or description that's ok too. Just let me know and I'll get it merged to make sure you get credit for Hacktoberfest. We appreciate the contribution!

File: angular/projects/spark-angular/src/lib/components/sprk-modal/sprk-modal.component.ts (hunk 1 above)
Comment (ka6thick54) [reply angular/projects/spark-angular/src/lib/components/sprk-modal/sprk-modal.component.ts:136]: I ll do the test and description.

**Comment by ka6thick54** [comment]
@guern1kn I added the description and test cases as requested. Let me know your thoughts.

**Review by guern1kn — APPROVED** [review]
Thank you for your contribution, @ka6thick54 ! Looking forward to more!

**Pull request merged**